CRM/Utils/Array.php

```diff
@@ -1058,16 +1058,20 @@ public static function formatArrayKeys(&$array) {
     if (!is_array($array)) {
       return;
     }
-    $keys = array_keys($array, 1);
-    if (count($keys) > 1 ||
-      (count($keys) == 1 &&
-        (current($keys) > 1 ||
-          is_string(current($keys)) ||
-          (current($keys) == 1 && $array[1] == 1) // handle (0 => 4), (1 => 1)
-        )
-      )
-    ) {
-      $array = $keys;
+
+    $hasSequentialKeysStartingWithZero = array_keys($array) === range(0, count($array) - 1);
+    $hasAllValuesAsNumberOne = array_unique(array_values($array)) === ['1'];
+    $hasMoreThanOneElement = count($array) > 1;
+    $isAssociative = !$hasSequentialKeysStartingWithZero || ($hasMoreThanOneElement && $hasAllValuesAsNumberOne);
```

Comment:
if you decided not to create the above method and keep things as is then $isAssociative is a misleading name, maybe $requireFormat is a more clear one ?

Reply:
I try to call variables exactly what they keep, otherwise you may ask "why does it then require format?". Ideally:

```php
$isSequential = ...;
$requiresFormatting = !$isSequential;
```